.github/scripts/functions.sh

```diff
@@ -0,0 +1,121 @@
+#!/bin/bash
+
+failed=0
+total=0
+
+# 1: file, 2: regex, 3: method_to_find
+function check_regex () {
+    (( total++ ))
+
+    grep -zoP "$2" $1 > /dev/null
+
+    if [ ! $? -eq 0 ]; then
+        (( failed++ ))
+        echo "❌ $3 is not implemented"
+        return 1
+    fi
+
+    echo "✅ $3 is implemented"
+    return 0
+}
+
+# 1: file
+function check_shared_ptr_copy_constructor() {
+    NAME='Copy constructor'
+    PATTERN='shared_ptr\s*\(\n?\s*?((const shared_ptr)|(shared_ptr const))&[^&]*?\)'
```

Comment:
Do not put spaces in regex for C++ code :) `\s+` is more generic and can behave properly if someone put an additional space.
`\s` already contain `\n`, there is no need to have it in a regex. 
`*?` doesn't make sense. `*` means any number, including 0.
```suggestion
    PATTERN='shared_ptr\s*\(\s*((const\s+shared_ptr)|(shared_ptr\s+const))\s*&.*\)'
```
Please check if my changes still pass the tests.
You can test your regex using https://regex101.com